src/App.tsx

```diff
@@ -15,36 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+function getPrepareGoods(
+  goods: string[],
+  sortBy: SortType,
+  isReversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setReversed] = useState(false);
+  const sortedGoods = getPrepareGoods(goodsFromServer, sortType, isReversed);
+
+  const resetReason = sortType || isReversed;
+
+  const handleReset = () => {
+    setSortType(SortType.Default);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== SortType.Alphabet,
+          })}
+          onClick={() => setSortType(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortType !== SortType.Length,
+          })}
+          onClick={() => setSortType(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(isReversed => !isReversed)}
```

Comment:
```suggestion
          onClick={() => setReversed(prevState => !prevState)}
```

Because linter doesn't allow it